cli/src/program.rs

```diff
@@ -2253,6 +2315,8 @@ fn do_process_program_write_and_deploy(
             min_rent_exempt_program_data_balance,
         )
     };
+
+    initial_instructions.extend(ixs);
```

Comment:
It could error out when making `ixs` above, so maybe more economic to extend `ComputeBudgetInstruction::set_compute_unit_price(...)` into `ixs`, instead of allocate set_compute_unit_price then fails on rpc call.

Reply:
After reading this https://www.helius.dev/blog/priority-fees-understanding-solanas-transaction-fee-mechanics Section:Best practices , it says set_compute_limit and set_compute_units should be always added to the start of ix Vec.

So a solution for this can be i create set_compute_units and set_compute_limit instructions after `ixs` is prepared ?